test/ranges-test.cc

```diff
@@ -213,7 +213,8 @@ TEST(ranges_test, tuple_parse_calls_element_parse) {
   EXPECT_THROW(f.parse(ctx), bad_format);
 }
 
-#ifdef FMT_RANGES_TEST_ENABLE_FORMAT_STRUCT
+#if defined(FMT_RANGES_TEST_ENABLE_JOIN) || \
+    defined(FMT_RANGES_TEST_ENABLE_FORMAT_STRUCT)
```

Comment:
We can probably remove these macros - they are testing for old variants of MSVC <= 2017.

Reply:
``struct tuple_like`` is used in tests that are enabled if these macros are defined.
I think enabling tests for early MSVC is out of scope of this PR.

Reply:
As I see MSVS 2015 is no longer supported (https://github.com/fmtlib/fmt/commit/8ed4a9dcc1c04cfaedcddc3ac25918cf35250f4a).
I propose to remove these checks in a separate PR.

What are the minimum compilers versions required for fmt now?

Reply:
For MSVC we definitely don't support anything older than 2017. 2017 is a bit tricky because there are multiple variants of it with older ones being quite broken and probably not worth supporting. Separate PR sounds reasonable.